mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulGameScreen.kt

```diff
@@ -1,59 +1,129 @@
+import ch.epfl.sdp.mobile.application.chess.engine.Color.White
+import ch.epfl.sdp.mobile.application.chess.notation.serialize
 import ch.epfl.sdp.mobile.state.SnapshotChessBoardState.SnapshotPiece
 import ch.epfl.sdp.mobile.ui.game.ChessBoardState
 import ch.epfl.sdp.mobile.ui.game.GameScreen
 import ch.epfl.sdp.mobile.ui.game.GameScreenState
-import ch.epfl.sdp.mobile.ui.game.Move
+import ch.epfl.sdp.mobile.ui.game.GameScreenState.Message
+import ch.epfl.sdp.mobile.ui.game.GameScreenState.Message.*
+import ch.epfl.sdp.mobile.ui.game.GameScreenState.Move
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
 
+/**
+ * The different navigation actions which may be performed by the [StatefulGameScreen].
+ *
+ * @param onBack the action to perform when going back.
+ * @param onShowAr the action to perform when AR should be started for the match.
+ */
+data class StatefulGameScreenActions(
+    val onBack: () -> Unit,
+    val onShowAr: (Match) -> Unit,
+)
+
 /**
  * The [StatefulGameScreen] to be used for the Navigation
  *
  * @param user the currently logged-in user.
  * @param id the identifier for the match.
- * @param modifier the [Modifier] for the composable
+ * @param actions the [StatefulGameScreenActions] to perform.
+ * @param modifier the [Modifier] for the composable.
+ * @param paddingValues the [PaddingValues] for this composable.
  */
 @Composable
 fun StatefulGameScreen(
     user: AuthenticatedUser,
     id: String,
+    actions: StatefulGameScreenActions,
     modifier: Modifier = Modifier,
+    paddingValues: PaddingValues = PaddingValues(),
 ) {
   val facade = LocalChessFacade.current
   val scope = rememberCoroutineScope()
   val match = remember(facade, id) { facade.match(id) }
 
-  val gameScreenState = remember(user, match, scope) { SnapshotChessBoardState(user, match, scope) }
+  val gameScreenState =
+      remember(actions, user, match, scope) {
+        SnapshotChessBoardState(
+            actions = actions,
+            user = user,
+            match = match,
+            scope = scope,
+        )
+      }
 
-  GameScreen(gameScreenState, modifier)
+  GameScreen(
+      state = gameScreenState,
+      modifier = modifier,
+      contentPadding = paddingValues,
+  )
 }
 
 /**
  * An implementation of [GameScreenState] that starts with default chess positions, can move pieces
- * and has a static move list
+ * and has a static move list.
+ *
+ * @param actions the actions to perform when navigating.
+ * @param user the currently authenticated user.
+ * @param match the match to display.
+ * @param scope a [CoroutineScope] keeping track of the state lifecycle.
  */
 class SnapshotChessBoardState(
+    private val actions: StatefulGameScreenActions,
     private val user: AuthenticatedUser,
     private val match: Match,
     private val scope: CoroutineScope,
 ) : GameScreenState<SnapshotPiece> {
 
+  // TODO : Implement these things.
```

Comment:
I actually intentionally kept these comments since voice recognition isn't actually implemented at the moment, while the rest of the `SnapshotChessBoardState` is functional. It's also why these methods and fields are at the top of the class.